department-of-veterans-affairs/caseflow
# Create a new Link component in Caseflow (based on the component from the Frontend Toolkit)

Connects #15173

### Description
While investigating the Reader bug described in the linked ticket, it came to light that the [Link](https://github.com/department-of-veterans-affairs/caseflow-frontend-toolkit/blob/master/components/Link.jsx) component in the Frontend Toolkit had been the source of problematic behavior in the past per @jcq, and it was thought that perhaps relocating the component to reside within Caseflow might be beneficial.  The exact reason for the issue is unclear, and should we decide to keep the component within Caseflow there should most likely be a follow-up ticket to remove the component from the FE toolkit altogether.

### Acceptance Criteria

### Testing Plan 
(note: it will be necessary for this to be merged into master and in Caseflow before it can be determined if this is a fix)
1. go to `localhost:3000` and login as any user with Reader access e.g. `BVAAABSHIRE`
2. under `App Selector` select the `Reader` product and the bulleted `Document List` underneath 
3. Select any of the available documents
4. Ensure the document loads correctly and you are able to use the `Previous` and `Next` buttons in the view to select different documents
5. Use the `Back` button in the UI (not the browser's built in back button) to return to the list of documents
6. Ensure the documents load properly and that the page/component does not reload unnecessarily 
7. User the search bar to search for a particular `Document Type`, e.g. Correspondence
8. Utilize one of the `Categories` filters as well
9. Select one of the documents from the search results and repeat steps 4 - 6
10. When returning to the document list ensure that the search results and Categories filtering are preserved
